### ISSUE TITLE
(groups || []).concat is not a function

### ISSUE THREAD

**Issue by jgrammen-agilitypr** [Alexandre-io/verdaccio-ldap#35]
root@npm:~# nodejs -v
v10.3.0

-- verdaccio-ldap@2.0.0
+-- verdaccio@3.0.2

ubuntu 16.04

everytime I authenticate it crashes the verdaccio service. with the following error in the log:
```
{"name":"verdaccio","hostname":"npm","pid":14503,"level":60,"err":{"message":"(groups || []).concat is not a function","name":"TypeError","stack":"TypeError: (groups || []).concat is not a function\n    at authenticatedUser (/usr/lib/node_modules/verdaccio/build/lib/auth.js:372:32)\n    at /usr/lib/node_modules/verdaccio/build/lib/auth.js:105:26\n    at sendResult (/usr/lib/node_modules/verdaccio-ldap/node_modules/ldapjs/lib/client/client.js:1395:12)\n    at messageCallback (/usr/lib/node_modules/verdaccio-ldap/node_modules/ldapjs/lib/client/client.js:1421:16)\n    at /usr/lib/node_modules/verdaccio-ldap/node_modules/ldapjs/lib/client/client.js:1282:14\n    at Array.forEach (<anonymous>)\n    at Client._onClose (/usr/lib/node_modules/verdaccio-ldap/node_modules/ldapjs/lib/client/client.js:1272:19)\n    at Object.onceWrapper (events.js:273:13)\n    at Socket.emit (events.js:182:13)\n    at TCP._handle.close [as _onclose] (net.js:596:12)"},"msg":"uncaught exception, please report this\n@{err.stack}","time":"2018-06-05T19:13:08.335Z","v":0}

```

```
auth:
  ldap:
    type: ldap
    client_options:
      url: "ldap://ca02vwdc01.agilitypr.net:389"
      # Only required if you need auth to bind
      adminDn: "CN=verdaccio,OU=Service Accounts,DC=domain,DC=net"
      adminPassword: "passwordhidden"
      # Search base for users
      searchBase: "OU=Users,OU=CORP,DC=domain,DC=net"
      searchFilter: "(sAMAccountName={{username}})"
      # If you are using groups, this is also needed
      groupDnProperty: 'cn'
      groupSearchBase: "OU=Users,OU=CORP,DC=domain,DC=net"
      # If you have memberOf support on your ldap
      searchAttributes: ['*', 'memberOf']
      # Else, if you don't (use one or the other):
      groupSearchFilter: '(memberUid={{dn}})'
      # Optional
      cache: False

```

**Comment by juanpicado** [comment]
Duplicate https://github.com/Alexandre-io/verdaccio-ldap/issues/33

**Comment by jgrammen-agilitypr** [comment]
in #33 they user did not appear to have the concat error when running nodejs version 10.x, and that ticket appeared to about a whole bunch of issues.

If this ticket is a duplicate and not useful, then feel free to close it.
and I will watch the other ticket in anticipation of a fix.

**Comment by juanpicado** [comment]
I just marked it as a reference. Perhaps is different, but both fail in the same line in verdaccio.

**Comment by Alexandre-io** [comment]
Should be fixed since 2.1.0
